src/App.tsx

```diff
@@ -1,14 +1,77 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [isListLoaded, setIsListLoaded] = useState(false);
+  const [filterBy, setFilterBy] = useState('all');
+  const [inputFilter, setInputFilter] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [selectedUser, setSelectedUser] = useState<User>({
+    id: 0,
+    name: '',
+    email: '',
+    phone: '',
+  });
+  const [isModalLoaded, setIsModalLoaded] = useState(false);
+
+  useEffect(() => {
+    if (selectedTodoId !== 0) {
+      getUser(selectedTodoId)
+        .then(user => {
+          setSelectedUser(user);
+          setIsModalLoaded(true);
+        });
+    }
+  }, [selectedTodoId]);
+
+  useEffect(() => {
+    getTodos()
+      .then((todos: Todo[]) => {
+        setTodoList(todos);
+        setIsListLoaded(true);
+      });
+  }, []);
```

Comment:
Move these functions declarations from useEffect (But call them in useEffect). use useCallback for them. And try to rewrite it with using async/await.